Lib/types.py

```diff
@@ -165,14 +165,19 @@ class Baz(list[str]): ...
         assert get_original_bases(int) == (object,)
     """
     try:
-        return cls.__orig_bases__
+        cls_dict = cls.__dict__
```

Comment:
Yes, I agree that we don't really need to worry about such things here. This is a useful introspection function, but I don't think we're trying to make the same guarantees as `inspect.getattr_static()` about never triggering descriptor lookups etc.

In fact, I'm not sure we even need to worry about the possibility that a class doesn't have a `__dict__` attribute. (I'm not sure that's even possible? At the very least, none of the tests in this PR provide coverage for that branch.) I would just go for a somewhat simpler implementation:

```py
def get_original_bases(cls):
    try:
        return cls.__dict__.get("__orig_bases__", cls.__bases__)
    except AttributeError:
        raise TypeError(
            f"Expected an instance of type, not {type(cls).__name__!r}"
        ) from None
```

Reply:
`type_dict` in `Objects/typeobj.c` does have a case where it returns None, but I think that may be dead code; other adjacent code blindly assumes it can get a `__dict__` from any type.

@Gobot1234 could you apply Alex's suggested change?

Reply:
Yep, done